pkg/idp/provider/azure/secret.go

```diff
@@ -0,0 +1,89 @@
+package azure
+
+import (
+	"strings"
+	"time"
+
+	"github.com/dexidp/dex/connector/microsoft"
+	"github.com/giantswarm/microerror"
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+	"gopkg.in/yaml.v2"
+)
+
+type azureSecret struct {
```

Comment:
Also adding a struct for the secret since returning a bunch of strings is messy and error prone.